mailbox/store/src/test/java/org/apache/james/mailbox/store/StoreMailboxManagerTest.java

```diff
@@ -166,5 +176,34 @@ public void getMailboxShouldThrowWhenMailboxDoesNotMatchUser() throws Exception
 
         assertThat(expected.getId()).isEqualTo(MAILBOX_ID);
     }
+
+    @Test
+    public void loginShouldCreateSessionWhenGoodPassword() throws Exception {
+        MailboxSession expected = storeMailboxManager.login(CURRENT_USER, CURRENT_USER_PASSWORD, LOGGER);
+
+        assertThat(expected.getUser().getUserName()).isEqualTo(CURRENT_USER);
+    }
+
+    @Test(expected = BadCredentialsException.class)
+    public void loginShouldThrowWhenBadPassword() throws Exception {
+        storeMailboxManager.login(CURRENT_USER, "badpassword", LOGGER);
+    }
+
+    @Test(expected = BadCredentialsException.class)
+    public void loginAsOtherUserShouldNotCreateUserSessionWhenAdminWithBadPassword() throws Exception {
+        storeMailboxManager.loginAsOtherUser(ADMIN, "badpassword", CURRENT_USER, LOGGER);
+    }
+
+    @Test(expected = BadCredentialsException.class)
```

Comment:
I believe this is the wrong exception. 

We should have a "not admin" exception.